app/views/events/index.html.erb

```diff
@@ -0,0 +1,14 @@
+<% provide(:title, "All Events") %>
+
+<h1>All Events</h1>
+
+Past Events:<P>
+
+<% Event.past.each do |e| %>
+	<%= link_to e.name, e %>, created by <%= e.creator.name%> <p>
+<% end %>
+<p>Upcoming Events:<p>
+
+<% Event.upcoming.each do |e| %>
+	<%= link_to e.name, e %>, created by <%= e.creator.name%> <p>
+<% end %>
```

Comment:
Please remove this (`Events.past` and `Events.upcoimng`) logic from the view and create:

 - [ ] `@past_events = Event.past` to the Events' controller index action.
 - [ ] `@upcoming_events = Event.upcoming` to the Events' controller index action.

Then you can conveniently rely on `@past_events` and `@upcoming_events` in your view. This is the cleaner way (Check out [this](https://www.toptal.com/ruby-on-rails/top-10-mistakes-that-rails-programmers-make#common-mistake-2-putting-too-much-logic-in-the-view) about putting too much logic in the view)